DataBiosphere/leonardo
# [IA-4291] use Wsm DB to decide if runtime should be deleted

__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA4291




## Summary of changes


### What

When we decide whether to send the wsmResouceId to back leo (which determines if a delete call to WSM is made), this changes the logic to use the WSM java client to check the WSM DB as the source of truth.

### Why

Before we were relying on Leo's database and checking if the hostIp was set, but this was an unreliable metric to decide if the wsmResource should be deleted.


## Testing these changes

### What to test

- Create an azure VM
- Delete your vm
- check that it was deleted in the Azure portal and/or examine to logs to ensure there was no "no-op for wsm.deleteVm" message



### Who tested and where

  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
